app-o11y-jaeger-agent-mixin/jaeger.libsonnet

```diff
@@ -0,0 +1,21 @@
+local k = import 'ksonnet-util/kausal.libsonnet';
+
+{
+  _config+:: {
+    cluster: error 'Must define a cluster',
+    namespace: error 'Must define a namespace',
+    jaeger_agent_host: null,
+  },
+
+  local container = k.core.v1.container,
+
+  jaeger_mixin::
+    if $._config.jaeger_agent_host == null
+    then {}
+    else
+      container.withEnvMixin([
+        container.envType.new('JAEGER_AGENT_HOST', $._config.jaeger_agent_host),
+        container.envType.new('JAEGER_TAGS', 'namespace=%s,service.namespace=%s,cluster=%s' % [$._config.namespace, $._config.namespace, $._config.cluster]),
```

Comment:
Rather than producing a new mixin, the existent one could have a feature flag to allow folks opting it at their own pace. This has the advantage of making the mixin backwards compatible plus folks won't need to vendor a new library.

You could extract the _value_ part of `JAEGER_TAGS` to a local and have conditional logic to decide what you introduce there.

Something like:

```jsonnet
local k = import 'ksonnet-util/kausal.libsonnet';

{
  _config+:: {
    cluster: error 'Must define a cluster',
    namespace: error 'Must define a namespace',
    jaeger_agent_host: null,
    with_service_namespace: false,
  },

  local container = k.core.v1.container,

  jaeger_mixin::
    if $._config.jaeger_agent_host == null
    then {}
    else
      local jaegerTags = if $._config.with_service_namespace then
        'namespace=%s,service.namespace=%s,cluster=%s' % [$._config.namespace, $._config.namespace, $._config.cluster]
      else
        'namespace=%s,cluster=%s' % [$._config.namespace, $._config.cluster];
      container.withEnvMixin([
        container.envType.new('JAEGER_AGENT_HOST', $._config.jaeger_agent_host),
        container.envType.new('JAEGER_TAGS', jaegerTags),
        container.envType.new('JAEGER_SAMPLER_MANAGER_HOST_PORT', 'http://%s:5778/sampling' % $._config.jaeger_agent_host),
      ]),
}

```

Reply:
Thanks for that suggestion - I'll give it a go !